thegraphnetwork-literev/es-journals
# feat: Implement CI test configuration

This Pull Request introduces a series of enhancements to our Continuous Integration (CI) pipeline and development environment configuration, aiming to improve the robustness, flexibility, and maintainability of our project setup. It incorporates automated tests for health checks, adapts the Conda environment activation across different environments, separates tasks for testing certificates, and updates the Conda environment setup for development purposes.

**Key Highlights:**
- Implemented automated health check tests to ensure the services are up and running before proceeding with further CI steps.
- Modified the script to activate the Conda environment dynamically, catering to various environments and ensuring compatibility across different setups.
- Created a separate task for certificate testing.
- Updated the Miniconda version in our CI pipeline to leverage the latest features and fixes, ensuring our environment is up-to-date.
- Implemented CI test configurations, laying the groundwork for comprehensive automated testing within our CI pipeline.
- Updated the Conda environment and added a development group on containers-sugar for deploying Elasticsearch in development mode.

## How to Test These Changes

Replicate the CI steps locally by running the corresponding scripts and commands to verify the setup and tests work in your local development environment.

## Pull Request Checklists

This PR is a:

About this PR:
- [x] the tests generate log file(s) (not applicable).

Author's checklist:

## Additional Information

This PR lays a solid foundation for future developments, ensuring our CI pipeline and development environment are both scalable and maintainable. It addresses several pain points previously encountered in our development workflow and sets the stage.

## Reviewer's Checklist

```
## Reviewer's Checklist

```
